src/postings/block_segment_postings.rs

```diff
@@ -265,22 +293,24 @@ impl BlockSegmentPostings {
     pub(crate) fn shallow_seek(&mut self, target_doc: DocId) {
         if self.skip_reader.seek(target_doc) {
             self.block_max_score_cache = None;
+            self.block_loaded = false;
         }
     }
 
     pub(crate) fn block_is_loaded(&self) -> bool {
-        self.loaded_offset == self.skip_reader.byte_offset()
+        self.block_loaded
     }
 
     pub(crate) fn load_block(&mut self) {
         let offset = self.skip_reader.byte_offset();
-        if self.loaded_offset == offset {
+        // I'm not sure why it's there, but it breaks with blocks that can be encoded on zero byte.
```

Comment:
```suggestion
        // I'm not sure why it's there, but it breaks with blocks that can be encoded on zero byte.
```

do we still need the comment?